community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ImportLogic.java

```diff
@@ -0,0 +1,528 @@
+    private final Log log;
+    private final ExecutionMonitor executionMonitor;
+    private final RecordFormats recordFormats;
+    private final CountingStoreUpdateMonitor storeUpdateMonitor = new CountingStoreUpdateMonitor();
+    private final long maxMemory;
+
+    // This map contains additional state that gets populated, created and used throughout the stages.
+    // The reason that this is a map is to allow for a uniform way of accessing and loading this stage
+    // from the outside. Currently these things live here:
+    //   - RelationshipTypeDistribution
+    private final Map<Class<?>,Object> accessibleState = new HashMap<>();
+
+    // components which may get assigned and unassigned in some methods
+    private NodeRelationshipCache nodeRelationshipCache;
+    private NodeLabelsCache nodeLabelsCache;
+    private long startTime;
+    private InputCache inputCache;
+    private NumberArrayFactory numberArrayFactory;
+    private Collector badCollector;
+    private IoMonitor writeMonitor;
+    private IdMapper idMapper;
+    private IdGenerator idGenerator;
+    private MemoryUsageStatsProvider memoryUsageStats;
+    private InputIterable<InputNode> nodes;
+    private InputIterable<InputRelationship> relationships;
+    private InputIterable<InputNode> cachedNodes;
+    private long peakMemoryUsage;
+    private long availableMemoryForLinking;
+
+    /**
+     * Advanced usage of the parallel batch importer, for special and very specific cases. Please use
+     * a constructor with fewer arguments instead.
+     *
+     * @param storeDir directory which the db will be created in.
+     * @param fileSystem {@link FileSystemAbstraction} that the {@code storeDir} lives in.
+     * @param neoStore {@link BatchingNeoStores} to import into.
+     * @param config import-specific {@link Configuration}.
+     * @param logService {@link LogService} to use.
+     * @param executionMonitor {@link ExecutionMonitor} to follow progress as the import proceeds.
+     * @param recordFormats which {@link RecordFormats record format} to use for the created db.
+     * @param input {@link Input} containing the data to import.
+     */
+    public ImportLogic( File storeDir, FileSystemAbstraction fileSystem, BatchingNeoStores neoStore,
+            Configuration config, LogService logService, ExecutionMonitor executionMonitor,
+            RecordFormats recordFormats, Input input )
+    {
+        this.storeDir = storeDir;
+        this.fileSystem = fileSystem;
+        this.neoStore = neoStore;
+        this.config = config;
+        this.recordFormats = recordFormats;
+        this.log = logService.getInternalLogProvider().getLog( getClass() );
+        this.executionMonitor = ExecutionSupervisors.withDynamicProcessorAssignment( executionMonitor, config );
+        this.maxMemory = config.maxMemoryUsage();
+
+        initialize( input );
+    }
+
+    private void initialize( Input input )
+    {
+        log.info( "Import starting" );
```

Comment:
we do not really start anything here right?

Reply:
Well, we set things up and the very next thing that happens is actual import stuff, so I think itäs a good place to have it